core/src/main/scala/kafka/zk/ZkData.scala

```diff
@@ -317,4 +310,4 @@ object AclChangeNotificationSequenceZNode {
   def deletePath(sequenceNode: String) = s"${AclChangeNotificationZNode.path}/${sequenceNode}"
   def encode(resourceName : String): Array[Byte] = resourceName.getBytes(UTF_8)
   def decode(bytes: Array[Byte]): String = new String(bytes, UTF_8)
```

Comment:
FWIW, `new String(bytes, "utf8")` is still more optimized than `new String(bytes, Charsets.UTF_8)` – see https://github.com/druid-io/druid/blob/master/java-util/src/main/java/io/druid/java/util/common/StringUtils.java#L32-L33
sounds like this PR is more focused on avoiding garbage, but if we have other places with lots of conversions still happening, this might be worth looking into

Reply:
That was fixed in recent versions.

Reply:
good to know, do you know which release?